### PR TITLE
LibWeb: Use is_ascii_case_insensitive_match() where the spec says to

### PR DESCRIPTION
**Pull request by AtkinsSJ**
It doesn't behave differently for now, but will do once we start using the new string types which use Unicode definitions of case sensitivity.

### CODE REVIEW

File: Userland/Libraries/LibWeb/DOM/Document.cpp [hunk 1]
```diff
@@ -1270,42 +1271,44 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<Event>> Document::create_event(DeprecatedSt
 
     // 2. If interface is an ASCII case-insensitive match for any of the strings in the first column in the following table,
     //      then set constructor to the interface in the second column on the same row as the matching string:
-    auto interface_lowercase = interface.to_lowercase();
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/DOM/Document.cpp:1273]: This one is unfortunate: We lose the caching of the lowercase string. Then again, we don't have to allocate a new String, so you win some and you lose some.

File: Userland/Libraries/LibWeb/DOM/Document.cpp [hunk 2]
```diff
@@ -1270,42 +1271,44 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<Event>> Document::create_event(DeprecatedSt
...
+    } else if (Infra::is_ascii_case_insensitive_match(interface, "devicemotionevent"sv)) {
         event = Event::create(realm, ""); // FIXME: Create DeviceMotionEvent
-    } else if (interface_lowercase == "deviceorientationevent") {
+    } else if (Infra::is_ascii_case_insensitive_match(interface, "deviceorientationevent"sv)) {
         event = Event::create(realm, ""); // FIXME: Create DeviceOrientationEvent
-    } else if (interface_lowercase == "dragevent") {
+    } else if (Infra::is_ascii_case_insensitive_match(interface, "dragevent"sv)) {
         event = Event::create(realm, ""); // FIXME: Create DragEvent
-    } else if (interface_lowercase.is_one_of("event", "events")) {
+    } else if (Infra::is_ascii_case_insensitive_match(interface, "evesvnt"sv)
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/DOM/Document.cpp:1286]: :mag_right: "evesvnt"

Comment (trflynn89) [reply Userland/Libraries/LibWeb/DOM/Document.cpp:1286]: Let's add a template helper? For cases like this where we compare to multiple things:

```c++
template<typename... Ts>
bool is_ascii_case_insensitive_match_to_one_of(StringView string, Ts&&... strings) 
{
    return (... || is_ascii_case_insensitive_match(string, forward<Ts>(strings)));
}
```

Comment (AtkinsSJ) [reply Userland/Libraries/LibWeb/DOM/Document.cpp]: Thank you!!! My multi-cursor editing is a bit funky sometimes.

File: Meta/Lagom/Tools/CodeGenerators/LibWeb/GenerateCSSMediaFeatureID.cpp [hunk 3]
```diff
@@ -100,7 +102,7 @@ Optional<MediaFeatureID> media_feature_id_from_string(StringView string)
         member_generator.set("name", name);
         member_generator.set("name:titlecase", title_casify(name));
         member_generator.append(R"~~~(
-    if (string.equals_ignoring_case("@name@"sv))
+    if (Web::Infra::is_ascii_case_insensitive_match(string, "@name@"sv))
```
Comment (linusg) [inline Meta/Lagom/Tools/CodeGenerators/LibWeb/GenerateCSSMediaFeatureID.cpp:105]: Omit `Web::`

File: Meta/Lagom/Tools/CodeGenerators/LibWeb/GenerateCSSPropertyID.cpp [hunk 4]
```diff
@@ -189,7 +190,7 @@ PropertyID property_id_from_string(StringView string)
         member_generator.set("name", name);
         member_generator.set("name:titlecase", title_casify(name));
         member_generator.append(R"~~~(
-    if (string.equals_ignoring_case("@name@"sv))
+    if (Web::Infra::is_ascii_case_insensitive_match(string, "@name@"sv))
```
Comment (linusg) [inline Meta/Lagom/Tools/CodeGenerators/LibWeb/GenerateCSSPropertyID.cpp:193]: Omit `Web::`

**Review by linusg — CHANGES_REQUESTED** [review]

**Comment by AtkinsSJ** [comment]
Removed unnecessary `Web::` prefixes.

**Comment by ADKaster** [comment]
Somehow developed a conflict in Document.cpp

**Comment by AtkinsSJ** [comment]
Resolved the conflicts

**Pull request merged**